pubsub.go

```diff
@@ -843,10 +843,22 @@ func (p *PubSub) processLoop(ctx context.Context) {
 			}
 			preq.resp <- peers
 		case rpc := <-p.incoming:
-			p.handleIncomingRPC(rpc)
+			rpcs := []*RPC{rpc}
```

Comment:
you should keep a pre-allocated slice available so you don't need to
reallocated this slice on appends.